### ISSUE TITLE
Implement cross-entropy loss kernel

### ISSUE THREAD

**Issue by ypapadop-amd** [ypapadop-amd/ggml#125]
We need a cross entropy kernel implementation for the MNIST example; see https://github.com/ypapadop-amd/ggml/blob/7131ddcc8a9e8cf835d737baa295a6f2e2510efc/src/ggml-cpu/ops.cpp#L9876

The cross-entropy loss tests require a softmax kernel.
